ugvc/cnv/convert_cnv_results_to_vcf.py

```diff
@@ -0,0 +1,156 @@
+    
+    # Add ALT
+    header.add_line('##ALT=<ID=CNV,Description="Copy number variant region">')
+    header.add_line('##ALT=<ID=<DEL>,Description="Deletion relative to the reference">')
+    header.add_line('##ALT=<ID=<DUP>,Description="Region of elevated copy number relative to the reference">')
+    
+    # Add FILTER
+    header.add_line('##FILTER=<ID=PASS,Description="high confidence CNV call">')
+    header.add_line('##FILTER=<ID=UG-CNV-LCR,Description="">')
+    header.add_line('##FILTER=<ID=LEN,Description="">')
+
+    # Add INFO
+    header.add_line('##INFO=<ID=CONFIDENCE,Number=1,Type=String,Description="Confidence level for CNV call. 0-Low 1-High">')
+    header.add_line('##INFO=<ID=CopyNumber,Number=1,Type=Float,Description="copy number of CNV call">')
+    header.add_line('##INFO=<ID=RoundedCopyNumber,Number=1,Type=Integer,Description="rounded copy number of CNV call">')
+    #header.add_line('##INFO=<ID=END_POS,Description="end position of the CNV">')
+    header.add_line('##INFO=<ID=SVLEN,Number=.,Type=Integer,Description="CNV length">')
+    header.add_line('##INFO=<ID=SVTYPE,Number=1,Type=String,Description="CNV type. can be DUP or DEL">')
+    
+    # Add FORMAT
+    header.add_line('##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">')
+
+    # Open a VCF file for writing
+    if args.out_directory: 
+        out_directory=args.out_directory
+    else:
+        out_directory=""
+    outfile=pjoin(out_directory,sample_name+'.cnv.vcf')
+    
+    with pysam.VariantFile(outfile, mode='w', header=header) as vcf_out:
+        df_cnvs = pd.read_csv(args.cnv_annotated_bed_file,sep="\t",header=None)
+        df_cnvs.columns=['chr','start','end','info']
+        for index, row in df_cnvs.iterrows():
+            # Create a new VCF record
+            chrID = row['chr']
+            start = row['start']
+            end = row['end']
+            info = row['info']
+
+            CN = int(info.split("|")[0].replace("CN",""))
+            cnv_type = '<DUP>'
+            if CN<2:
+                cnv_type='<DEL>'
+
+            filters = []
+            for item in info.split(';'):
+                arr = item.split('|')
+                if len(arr)>1:
+                    filters.append(arr[1])
+            
+            record = vcf_out.new_record()
+            record.contig =  chrID
+            record.start = start
+            record.stop = end
+            record.ref = "N"
+            record.alts = (cnv_type,)
+
+            CONFIDENCE = 'HIGH'
+            if len(filters)>0 :
+                for f in filters:
+                    record.filter.add(f)
+                CONFIDENCE='MEDIUM'
+                if len(filters)>1 :
+                    CONFIDENCE='LOW'
+            else:
+                record.filter.add('PASS')
+          
+            record.info['CONFIDENCE'] = CONFIDENCE
+            record.info['CopyNumber'] = CN
+            record.info['RoundedCopyNumber'] = int(CN)
```

Comment:
I think that 'int()' always rounds down. Do you need to round down, or to the closest integer?

Reply:
closest integer...